safe-global/safe-wallet-web
# feat: Transaction status widget

## What it solves

Resolves #2022 

## How this PR fixes it
Implements the design of the right side "Transaction status" widget
Paints the icons according to the active step in the stepper

Adds the `ConfirmedProposedTx` modal into the `ModalProvider`

## Todo
- [x] pass the confirmation count to the stepper (might need to implement a review modal first)

## How to test it
1. Create a "Send Funds" transaction
2. Proceed to the 2nd step and observe the stepper progressively getting painted

### Test review screen
1. Click to sign a transaction from the queue
2. Observe the tx status widget show the Confirmations step painted with the number of confirmations

## Screenshots
<img width="1261" alt="Screenshot 2023-05-30 at 20 02 45" src="https://github.com/safe-global/safe-wallet-web/assets/32431609/d349bab8-29a1-4133-a8b4-76687435e6e1">
<img width="1270" alt="Screenshot 2023-05-30 at 20 02 38" src="https://github.com/safe-global/safe-wallet-web/assets/32431609/96911ddb-2c3d-4ee1-9bff-b376e814b35f">
<img width="982" alt="Screenshot 2023-05-31 at 12 54 25" src="https://github.com/safe-global/safe-wallet-web/assets/32431609/69fde4fe-089c-4340-9312-e2270676939a">


## Checklist
